server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -658,12 +667,18 @@ private IndexShard getIndexShard(final ShardId shardId) {
     final class ReroutePhase extends AbstractRunnable {
         private final ActionListener<Response> listener;
         private final Request request;
+        private final boolean executedByClient;
         private final ReplicationTask task;
         private final ClusterStateObserver observer;
         private final AtomicBoolean finished = new AtomicBoolean();
 
         ReroutePhase(ReplicationTask task, Request request, ActionListener<Response> listener) {
+            this(task, request, listener, false);
+        }
+
+        ReroutePhase(ReplicationTask task, Request request, ActionListener<Response> listener, boolean executedByClient) {
             this.request = request;
+            this.executedByClient = executedByClient;
```

Comment:
I prefer the `initiatedByNodeClient`  naming here with a comment explaining why  we have  that boolean